org-roam.el

```diff
@@ -1302,21 +1302,24 @@ If OLD-DESC is passed, and is not the same as the link
 description, it is assumed that the user has modified the
 description, and the description will not be updated. Else,
 update with NEW-DESC."
-  (save-excursion
-    (goto-char (point-min))
-    (while (re-search-forward org-link-any-re nil t)
-      (when-let ((link (org-element-lineage (org-element-context) '(link) t)))
-        (let ((type (org-element-property :type link))
-              (path (org-element-property :path link)))
-          (when (and (string-equal (expand-file-name path) old-path)
-                     (org-in-regexp org-link-bracket-re 1))
-            (let* ((label (if (match-end 2)
-                              (match-string-no-properties 2)
-                            (org-link-unescape (match-string-no-properties 1))))
-                   (new-label (if (string-equal label old-desc)
-                                  new-desc
-                                label)))
-              (replace-match (org-roam-format-link new-path new-label type)))))))))
+  (cl-flet ((get-new-link ()
```

Comment:
I don't think we need a cl-flet here. I think `org-in-regexp` butchers the old match data. Simpler to just wrap the `when-let` clause in a `save-match-data`, without introducing it as a function.